### PR TITLE
Match file extensions against blacklist to avoid multiple rebuilds tr…

### PR DESCRIPTION
**Pull request by mxdamien**
…iggered by temporary files

see #29

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -139,27 +141,37 @@ fn main() -> anyhow::Result<()> {
...
-                        panic!("Error reloading model: {:?}", err);
-                    }
-                };
-
-                // This will panic, if the other end is disconnected, which is
-                // probably the result of a panic on that thread, or the
-                // application is being shut down.
-                //
-                // Either way, not much we can do about it here, except maybe to
-                // provide a better error message in the future.
```
Comment (hannobraun) [inline src/main.rs:161]: You're removing this comment, even though it still applies in the changed version. Can you please re-add it?

File: src/main.rs [hunk 2]
```diff
@@ -139,27 +141,37 @@ fn main() -> anyhow::Result<()> {
...
+
+                let black_list = HashSet::from([
+                    OsStr::new("swp"),
+                    OsStr::new("tmp"),
+                    OsStr::new("swx"),
+                ]);
+
+                if (file_ext.is_some()
+                    && black_list.get(file_ext.unwrap()).is_none())
+                    | file_ext.is_none()
```
Comment (hannobraun) [inline src/main.rs:158]: This is just a nit-pick. Feel free to act on it (or comment, if you disagree), or just leave it as-is, as you prefer. It doesn't matter a whole lot.

That `file_ext.is_some()` followed by `file_ext.unwrap()` is a bit icky. `if let Some(ext) = file_ext` would be more robust, but can't be used as a direct replacement here.

Alternative approach:
``` rust
if let Some(ext) = file_ext {
    if black_list.contains(ext) {
        return;
    }
}

// insert rest of event handling here
```

Aside from using the more robust `if let`, this also makes the handling of the error case a self-contained thing, thereby making the code a bit easier to understand. It also saves one level of indentation for the happy path. But as I said, not that important, feel free to ignore.

**Review by hannobraun — CHANGES_REQUESTED** [review]
Thank you for the pull request, @mxdamien!

I found one problem (code comment went missing; see the review comment I left) and left another note to highlight a possible improvement.

File: src/main.rs (hunk 2 above)
Comment (mxdamien) [reply src/main.rs]: Thanks for the feedback.

Removing the comment was an accident - sorry for that.

The alternative approach indeed looks neater. Still getting used to the rust coding style coming from C/C++.
I've pushed the changes

Comment (hannobraun) [reply src/main.rs]: Hey, no problem, that what reviews are for! :smile:

**Review by hannobraun — APPROVED** [review]
Thank you for the changes, @mxdamien. Looks great!

**Pull request merged**